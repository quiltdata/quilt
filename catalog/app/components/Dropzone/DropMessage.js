import cx from 'classnames'
import * as React from 'react'
import * as M from '@material-ui/core'

const useStyles = M.makeStyles((t) => ({
  root: {
    ...t.typography.body2,
    alignItems: 'center',
    display: 'flex',
    flexGrow: 1,
    justifyContent: 'center',
    minHeight: t.spacing(4),
    paddingBottom: t.spacing(1),
    paddingTop: t.spacing(1),
    textAlign: 'center',
  },
  warning: {
    color: t.palette.warning.dark,
  },
  error: {
    color: t.palette.error.main,
  },
}))

export default function DropMessage({ disabled, error, warning }) {
  const classes = useStyles()

  const label = React.useMemo(() => {
    if (error) return error
    if (warning) return warning
    if (disabled) return ''
    return 'Drop files here or click to browse'
  }, [disabled, error, warning])

  return (
    <div
      className={cx(classes.root, {
        [classes.error]: !!error,
        [classes.warning]: !!warning,
      })}
    >
      {label}
    </div>
  )
}
