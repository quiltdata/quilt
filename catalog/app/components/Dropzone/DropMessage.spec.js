import * as React from 'react'
import renderer from 'react-test-renderer'

import DropMessage from './DropMessage'

describe('components/Dropzone/DropMessage', () => {
  it('should render with default message', () => {
    const tree = renderer.create(<DropMessage />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render error', () => {
    const tree = renderer
      .create(<DropMessage error="This is errror" warning="This is warning" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render warning', () => {
    const tree = renderer.create(<DropMessage warning="This is warning" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should be empty when disabled', () => {
    const tree = renderer
      .create(<DropMessage disabled error="This is errror" warning="This is warning" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
