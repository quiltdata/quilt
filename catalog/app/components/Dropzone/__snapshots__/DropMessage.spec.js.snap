// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Dropzone/DropMessage should be empty when disabled 1`] = `
<div
  className="makeStyles-root-1 makeStyles-error-3 makeStyles-warning-2"
>
  This is errror
</div>
`;

exports[`components/Dropzone/DropMessage should render error 1`] = `
<div
  className="makeStyles-root-1 makeStyles-error-3 makeStyles-warning-2"
>
  This is errror
</div>
`;

exports[`components/Dropzone/DropMessage should render warning 1`] = `
<div
  className="makeStyles-root-1 makeStyles-warning-2"
>
  This is warning
</div>
`;

exports[`components/Dropzone/DropMessage should render with default message 1`] = `
<div
  className="makeStyles-root-1"
>
  Drop files here or click to browse
</div>
`;
